src/code/operation.js

```diff
@@ -242,14 +253,37 @@ export default class Operation {
         this.anchors = this.anchors.filter(function(listAnchor) {
             return listAnchor !== originalPortal.id;
         });
-        this.addAnchor(newPortal)
-        for (let link_ in this.links) {
+        this.addAnchor(newPortal);
+        let linksToRemove = [];
+        for (let link_ = 0; link_ < this.links.length; link_++) {
```

Comment:
You can use this.links.forEach here.  It would look like this:
```
self = this;
this.links.forEach((link) => {
    // Do stuff.
    // Bear in mind that the arrow function binds the this variable so you can use self to refer to the Operation instance.
}
```

If you do this, have in mind that